server/aws-lsp-codewhisperer/src/language-server/chat/chatController.ts

```diff
@@ -81,21 +81,23 @@ export class ChatController implements ChatHandlers {
         let response: GenerateAssistantResponseCommandOutput
 
         try {
-            this.#log(
-                'Request from tab:',
-                params.tabId,
-                'conversation id:',
-                requestInput.conversationState?.conversationId ?? 'undefined'
-            )
+            this.#log('Request from tab:', params.tabId, 'conversation id:', session.sessionId ?? 'New session')
 
             response = await session.generateAssistantResponse(requestInput)
             this.#log('Response to tab:', params.tabId, JSON.stringify(response.$metadata))
         } catch (err) {
-            this.#log(`Q api request error ${err instanceof Error ? err.message : 'unknown'}`)
+            const authFollowType = getAuthFollowUpType(err)
 
+            if (authFollowType) {
+                this.#log(`Q auth error: ${getErrorMessage(err)}`)
+
+                return createAuthFollowUpResult(authFollowType)
+            }
+
+            this.#log(`Q api request error ${err instanceof Error ? err.message : 'unknown'}`)
             return new ResponseError<ChatResult>(
```

Comment:
Is my following understanding correct: The expectation in this error case is that the IDE extension uses the `sendError` contract of the chat-client to show an error on the chat-ui

Reply:
Yes, that is correct